detect.go

```diff
@@ -28,17 +30,29 @@ type ProjectParser interface {
 	NodeIsRequired(path string) (bool, error)
 }
 
-func Detect(buildpackYMLParser BuildpackConfigParser, configParser ConfigParser, projectParser ProjectParser) packit.DetectFunc {
+func Detect(buildpackYMLParser BuildpackConfigParser, configParser ConfigParser, projectParser ProjectParser, logger scribe.Logger) packit.DetectFunc {
 	return func(context packit.DetectContext) (packit.DetectResult, error) {
-		root := context.WorkingDir
+		var projectPath string
+		var ok bool
+		var err error
 
-		path, err := buildpackYMLParser.ParseProjectPath(filepath.Join(context.WorkingDir, "buildpack.yml"))
-		if err != nil {
-			return packit.DetectResult{}, fmt.Errorf("failed to parse buildpack.yml: %w", err)
+		if projectPath, ok = os.LookupEnv("BP_DOTNET_PROJECT_PATH"); !ok {
+			projectPath, err = buildpackYMLParser.ParseProjectPath(filepath.Join(context.WorkingDir, "buildpack.yml"))
+			if err != nil {
+				return packit.DetectResult{}, fmt.Errorf("failed to parse buildpack.yml: %w", err)
+			}
+
+			if projectPath != "" {
+				nextMajorVersion := semver.MustParse(context.BuildpackInfo.Version).IncMajor()
+				logger.Subprocess("WARNING: Setting the project path through buildpack.yml will be deprecated soon in Dotnet Execute Buildpack v%s.", nextMajorVersion.String())
```

Comment:
Given that logging during detection is a) silenced by default by `pack` and b) prints in a nondeterministic order based on all the detecting buildpacks running in parallel, I think it'd be a bit clearer to log this warning during the build phase.

Granted, it's a bit weird to parse the `buildpack.yml` during build just to log a warning, but I think that'll provide a clearer UX and doesn't carry a significant performance penalty.